src/Type/Php/CallUserFuncDynamicFunctionReturnTypeExtension.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\Type;
+
+class CallUserFuncDynamicFunctionReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return in_array(
+			$functionReflection->getName(),
+			[
+				'call_user_func',
+				'call_user_func_array',
+			],
+			TRUE
+		);
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		$args = $functionCall->args;
+
+		if (count($args) === 0) {
+			return $functionReflection->getVariants()[0]->getReturnType();
```

Comment:
This will not work for some edge cases around functions with multiple variants. Best to call `ParametersAcceptorSelector::selectFromArgs()`.

Reply:
Could you drop just few words on what's the significance/meaning/purpose of that? Practically what should I pass as `$parametersAcceptors`? What I am trying to do with that call?

Reply:
Some weird PHP functions like `strtok` have multiple variants, basically function overloading. If there's a function that would have different return types based on which variant gets chosen, then the logic that always selects `[0]` would be wrong.

`$parametersAcceptors` is `$functionReflection->getVariants()`